clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -692,6 +692,13 @@ def warn_maybe_falloff_nonvoid_function : Warning<
 def warn_falloff_nonvoid_function : Warning<
   "non-void function does not return a value">,
   InGroup<ReturnType>;
+def warn_const_attr_with_pure_attr : Warning<
+  "'const' attribute imposes more restrictions, 'pure' attribute ignored">,
+  InGroup<IgnoredAttributes>;
```

Comment:
nit: generally we prefer one patch does one thing manner. So maybe we can add this in later patches.

Reply:
But it is same case - incorrect 'pure' usage

Reply:
While this is not required, **incorrect** is too broad. e.g., a lot of clang's change  can be summarized as fixing incorrect  implementation for C++.

Reply:
I think the changes made here are sufficiently closely related, they are definitely 'near the line', but are sufficiently close (in that it is fixing up teh diagnostics for the pure/const attributes).